### ISSUE TITLE
Fix distutils.tests.test_register.RegisterTestCase.test_check_metadata_deprecated

### ISSUE THREAD

**Issue by terryjreedy** [python/cpython#115198]
When testing 3.11.8 *installed* on Win 10, the only failure is
```
FAIL: test_check_metadata_deprecated (distutils.tests.test_register.RegisterTestCase.test_check_metadata_deprecated)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Programs\Python311\Lib\distutils\tests\test_register.py", line 295, in test_check_metadata_deprecated
    self.assertEqual(len(w.warnings), 1)
AssertionError: 71 != 1
```
Running test.test_distutils, the warnings are
>C:\Programs\Python311\Lib\distutils\command\install.py:326:
  DeprecationWarning: check_home argument is deprecated and ignored.

and apparently 70 similar to
>C:\Programs\Python311\Lib\distutils\command\check.py:128: DeprecationWarning:
 The frontend.OptionParser class will be replaced by a subclass of argparse.ArgumentParser in Docutils 0.21 or later.

I don't know why this passes on CI and buildbots, but there are two easy fixes.

1. Change the test to `assertGreaterEqual(len(w.warnings), 1)`.
2. Remove the test.

@vstinner Which do you think would be better?

### Linked PRs
* gh-115200
* gh-115220

**Comment by serhiy-storchaka** [comment]
Please check whether #115200 helps.

**Comment by terryjreedy** [comment]
Yes (with changes).

**Comment by serhiy-storchaka** [comment]
The warnings are emitted by a newer version of Docutils. If Docutils is not installed, no warnings are emitted in this test and several other tests are skipped. According to them, this code will be broken in Docutils 0.21.

What should we do?

* Skip this test (and maybe several other tests) if docutils >= 0.21.
* Check the docutils version and raise more specific error for docutils >= 0.21.

But what if they will defer that change to later version? Testing this dynamically may be not easy because we do not know how exactly it will fail.

**Comment by serhiy-storchaka** [comment]
I looked in the docutils sources and try to fix it in other way: #115220.

**Comment by serhiy-storchaka** [comment]
On other hand, it is not compatible with the master branch of Docutils (future 0.21) due to other changes, so both PRs are equivalent: they fix tests with Docutils 0.19 and 0.20.

**Comment by terryjreedy** [comment]
Distutils is gone in 3.12.  Are we ever going to run distutils tests (which is to say, 3.11 tests) with a future docutils?  Are the docutils 0.21+ developers likely to do so, and care?  IE, can we wait until a 0.21 problem arises?  If we do wait, we could add a comment to `test_check_metadata_deprecated`: "# If this test breaks with docutils 0.21+, see issue 115198."
